admin_server/src/main/java/com/backend/admin_server/user_data/repository/UserRepository.java

```diff
@@ -1,20 +1,34 @@
 package com.backend.admin_server.user_data.repository;
 
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
 import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBScanExpression;
-import com.backend.admin_server.user_data.model.User;
-import org.springframework.beans.factory.annotation.Autowired;
+import com.backend.admin_server.user_data.model.UserModel;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
 
 @Repository
 public class UserRepository {
 
-    @Autowired
-    private DynamoDBMapper dynamoDBMapper;
+    private final DynamoDBMapper dynamoDBMapper;
 
-    public List<User> findAllUsers() {
-        return dynamoDBMapper.scan(User.class, new DynamoDBScanExpression());
+    public UserRepository(DynamoDBMapper dynamoDBMapper) {
+        this.dynamoDBMapper = dynamoDBMapper;
+    }
+
+    public List<UserModel> findAllUsers() {
+        return dynamoDBMapper.scan(UserModel.class, new DynamoDBScanExpression());
+    }
+
+    public UserModel findByUserId(Integer userId) {
+        UserModel userModel = new UserModel();
+        userModel.setUserId(userId);
+        DynamoDBQueryExpression<UserModel> queryExpression = new DynamoDBQueryExpression<UserModel>()
+                .withHashKeyValues(userModel);
+
+        return dynamoDBMapper.query(UserModel.class, queryExpression).stream()
```

Comment:
I did a dirty chatGPT question about this query, and I think we can use dynamoDBMapper.load(UserModel.class, userId); to retrieve the user since we're searching by primary key so only one user should exist. Let me know if I'm wrong though :)